subprojects/model-core/src/main/java/org/gradle/api/internal/provider/DefaultMapProperty.java

```diff
@@ -219,7 +219,12 @@ public void fromState(ExecutionTimeValue<? extends Map<? extends K, ? extends V>
         if (value.isMissing()) {
             setSupplier(noValueSupplier());
         } else if (value.isFixedValue()) {
-            setSupplier(new FixedSuppler<>(Cast.uncheckedNonnullCast(value.getFixedValue())));
+            SideEffect<?> sideEffect = value.getSideEffect();
+            if (sideEffect != null) {
+                setSupplier(new FixedWithSideEffectSupplier<>(Cast.uncheckedNonnullCast(value.getFixedValue()), Cast.uncheckedNonnullCast(sideEffect)));
+            } else {
+                setSupplier(new FixedSuppler<>(Cast.uncheckedNonnullCast(value.getFixedValue())));
```

Comment:
```suggestion
                setSupplier(new FixedSupplier<>(Cast.uncheckedNonnullCast(value.getFixedValue())));
```

I suppose there is a typo in the existing class.